modules/openthread/platform/crypto_psa_key_allocator.c

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) 2025 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "crypto_psa_key_allocator.h"
+
+#ifdef CONFIG_OPENTHREAD_PSA_NVM_BACKEND_KMU
+#include <cracen_psa_kmu.h>
+
+#define OPENTHREAD_KMU_SINGLE_KEY_SLOT_SIZE (2)
+/* Reserved for six 128-bit keys, and one 256-bit key */
+#define OPENTHREAD_KMU_DEDICATED_SLOTS	    (8)
+#endif /* CONFIG_OPENTHREAD_PSA_NVM_BACKEND_KMU */
+
+otError getKeyRef(otCryptoKeyRef *aInputKeyRef, psa_key_attributes_t *aAttributes)
+{
+	if (!aInputKeyRef) {
+		return OT_ERROR_INVALID_ARGS;
+	}
+
+#if defined(CONFIG_OPENTHREAD_PSA_NVM_BACKEND_ITS)
+	/* In this case we do not need make any adjustments to the input KeyRef or Attributes */
+	return OT_ERROR_NONE;
+
+#elif defined(CONFIG_OPENTHREAD_PSA_NVM_BACKEND_KMU)
+	/* Exit function if the key is outside the expected range dedicated to persistent lifetime.
+	 * The keys outside the range can be stored for example in volatile storage, so we should
+	 * simply exit this function with OT_ERROR_NONE.
+	 *
+	 * Currently we supports only 7 keys defined in the openthread/src/core/crypto/storage.hpp
+	 * file. The first six keys are 128-bit length (1 KMU slot), and the last one is 256-bit
+	 * length (2 kmu slots). If the OPENTHREAD_KMU_DEDICATED_SLOTS is changed the new
+	 * formula for converting keys must be written, because we must take into account the shift
+	 * of two slots for ECDSA private key (7th key).
+	 */
+	if (*aInputKeyRef <= CONFIG_OPENTHREAD_PSA_ITS_NVM_OFFSET ||
+	    *aInputKeyRef > CONFIG_OPENTHREAD_PSA_ITS_NVM_OFFSET + OPENTHREAD_KMU_DEDICATED_SLOTS) {
+		return OT_ERROR_NONE;
+	}
+
+	/* Convert key to KMU slot, currently all keys except ECDSA private key are 128-bit length
+	 * so need one KMU slot. The ECDSA private key is the last one on the list, so we can simply
+	 * convert it one by one. Keys starts from 1, so we need to decrease it by 1 to get the
+	 * correct slot.
+	 */
+	*aInputKeyRef = PSA_KEY_HANDLE_FROM_CRACEN_KMU_SLOT(
+		CRACEN_KMU_KEY_USAGE_SCHEME_RAW,
+		CONFIG_OPENTHREAD_KMU_SLOT_START +
+			(*aInputKeyRef - CONFIG_OPENTHREAD_PSA_ITS_NVM_OFFSET) - 1);
+
+	/* Convert key attributes to meet KMU requirements */
+	if (aAttributes) {
```

Comment:
is it okay if aAttributes is NULL? we don't need to do anything then and can return no error?